dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -69,9 +69,12 @@ inline void ParseSampleDesc(std::vector<SampleDesc>& samples_container,
   // Reading consecutive components
   ComponentDesc component;
   while (extensions_stream >> component.ext) {
-    DALI_ENFORCE(extensions_stream >> component.offset >> component.size,
-                 IndexFileErrMsg(index_path, line,
-                                 "size or offset corresponding to the extension not found"));
+    DALI_ENFORCE(
+        extensions_stream >> component.offset >> component.size >> component.filename,
```

Comment:
How about some backward compatibility?
```suggestion
        extensions_stream >> component.offset >> component.size;
        if (version >= 1002)  // or whatever way we have to represent the parsed version
          extensions_stream  component.filename,
```
Also, I guess it should be a `component_stream` now that we have more than just extensions there.

Reply:
Done